src/control-plane/backend/lambda/api/service/quicksight/reporting-utils.ts

```diff
@@ -263,11 +263,15 @@ export const retentionAnalysisVisualColumns: InputColumn[] = [
 
 export const createDataSet = async (quickSight: QuickSight, awsAccountId: string, principalArn: string,
   dataSourceArn: string,
-  props: DataSetProps)
+  props: DataSetProps, requestAction: ExploreRequestAction)
 : Promise<CreateDataSetCommandOutput|undefined> => {
 
   try {
-    const datasetId = uuidv4();
+    let datasetId = `${QUICKSIGHT_RESOURCE_NAME_PREFIX}${QUICKSIGHT_DATASET_INFIX}${uuidv4().replace(/-/g, '')}`;
+    if (requestAction === ExploreRequestAction.PREVIEW) {
+      datasetId = `${QUICKSIGHT_TEMP_RESOURCE_NAME_PREFIX}${datasetId}`;
```

Comment:
@llmin  make sure we can create dataset with such long id

Reply:
verified